sdk/storage/Azure.Storage.Blobs/tests/EncryptedBlockBlobClientTests.cs

```diff
@@ -18,25 +25,281 @@ public EncryptedBlockBlobClientTests(bool async)
         {
         }
 
-        // Placeholder test to verify things work end-to-end
-        [Test]
-        public async Task DeleteAsync()
+        #region Utility
+
+        /// <summary>
+        /// Gets a client to a nonexistent blob using client-side encryption in a brand new disposable container.
+        /// </summary>
+        /// <param name="blob">The blob client created.</param>
+        /// <returns>The IDisposable to delete the container when finished.</returns>
+        private IDisposable GetEncryptedBlockBlobClient(out EncryptedBlobClient blob, MockKeyEncryptionKey mock)
+        {
+            var disposable = this.GetNewContainer(out var container);
+            blob = new EncryptedBlobClient(
+                    new Uri(Path.Combine(container.Uri.ToString(), this.GetNewBlobName())), this.GetNewSharedKeyCredentials(),
+                    keyEncryptionKey: mock,
+                    keyResolver: mock);
+
+            return disposable;
+        }
+
+        // TODO this is a copy/paste. fix that.
+        /// <summary>
+        /// Gets and validates a blob's encryption-related metadata
+        /// </summary>
+        /// <param name="metadata">The blob's metadata</param>
+        /// <returns>The relevant metadata.</returns>
+        private EncryptionData GetAndValidateEncryptionData(IDictionary<string, string> metadata)
```

Comment:
Is this duplicated from the code?  Just make it internal and expose it via InternalsVisibleTo?